docs/DeveloperGuide.md

```diff
@@ -1,38 +1,584 @@
+
+Step 1. Given that the application already has existing data and there is one person being tracked, Alice, who belongs
+to the current generation. In this case `Family` would be initialised with one generation being tracked - myGen.
+
+![FindIncomeExpenditure1](images/FindIncomeExpenditure1.png)
+
+Step 2. The user executes `find /d Bills /c 1` command to search for income and expenditures with descriptions
+containing "Bills". The `find` command will be parsed and calls
+`Family#listExpenseSearch("Bills", 1)` and `Family#listSearch("Bills")` which would instantiate 2 temporary array list
+for storing the results of the upcoming search.
+
+![FindIncomeExpenditure2](images/FindIncomeExpenditure2.png)
+
+Step 3. After the temporary array list for expenditure has been created, the existing generation will be iterated
+for `Person` objects. The `expenditureList` for a person would be retrieved during that person's iteration
+and `MoneyList(temp)#searchExpense("Bills", 1)` will be called as `expenditureList` extends `MoneyList(temp)`. This
+method then iterates through the list and calls
+`Money(temp)#getCategory()` and `Money(temp)#getDescription()` on each expenditure, collecting and returning the
+expenditure if its category matches the given index and description matches the given description. The returned
+expenditures are then appended to the temporary array list.
+
+![FindIncomeExpenditure3](images/FindIncomeExpenditure3.png)
+
+Step 4. After the temporary array list for income has been created, the existing generation will be iterated
+for `Person` objects. The `incomeList` for a person would be retrieved during that person's iteration
+and `MoneyList(temp)#searchIncome("Bills")` will be called as `incomeList` extends `MoneyList(temp)`. This method then
+iterates through the list and calls
+`Money(temp)#getDescription()` on each income, collecting and returning the income if its description matches the given
+description. The returned incomes are then appended to the temporary array list.
+
+![FindIncomeExpenditure4](images/FindIncomeExpenditure4.png)
+
+Step 5. The iteration, collecting and appending to the temporary array list in step 3 and 4 is repeated until every
+person has been iterated. Finally, `Categories#getLabel(1)` is called so that an appropriate message can be displayed to
+the user, stating the name of the category, following by a series of print to display the expenditures in this category.
+
+> :information_source: **Note:** If the `index` provided does not map to any existing categories,
+> then it can be observed that there will never be any results returned. The `find` command will
+> check the index provided using `Parser#checkValidCategory` before iterating `Family`. If the check
+> fails, an error message will be displayed to the user instead of continuing with the execution.
+
+The following sequence diagram shows how the `find` operation works after the `FindCommand` has been created
```

Comment:
I think this sequence diagram may be too complicated? There seems to be too many classes in the diagram?
![image](https://user-images.githubusercontent.com/58723008/161183112-5b8cb035-34fb-4249-8455-32e8f48b7759.png)